src/special.f90

```diff
@@ -554,56 +555,68 @@ SUBROUTINE SPHJ(N,X,NM,SJ,DJ)
 !                MSTA1 and MSTA2 for computing the starting
 !                point for backward recurrence
 !       =======================================================
-!
-        IMPLICIT DOUBLE PRECISION (A-H,O-Z)
-        IMPLICIT integer (I-N)
-        DIMENSION SJ(0:N),DJ(0:N)
-        NM=N
-        IF (DABS(X).LT.1.0D-100) THEN
-           DO 10 K=0,N
-              SJ(K)=0.0D0
-10            DJ(K)=0.0D0
-           SJ(0)=1.0D0
-           IF (N.GT.0) THEN
-              DJ(1)=.3333333333333333D0
-           ENDIF
-           RETURN
-        ENDIF
-        SJ(0)=DSIN(X)/X
-        DJ(0)=(DCOS(X)-DSIN(X)/X)/X
-        IF (N.LT.1) THEN
-           RETURN
-        ENDIF
-        SJ(1)=(SJ(0)-DCOS(X))/X
-        IF (N.GE.2) THEN
-           SA=SJ(0)
-           SB=SJ(1)
-           M=MSTA1(X,200)
-           IF (M.LT.N) THEN
-              NM=M
-           ELSE
-              M=MSTA2(X,N,15)
-           ENDIF
-           F=0.0D0
-           F0=0.0D0
-           F1=1.0D0-100
-           DO 15 K=M,0,-1
-              F=(2.0D0*K+3.0D0)*F1/X-F0
-              IF (K.LE.NM) SJ(K)=F
-              F0=F1
-15            F1=F
-           CS=0.0D0
-           IF (DABS(SA).GT.DABS(SB)) CS=SA/F
-           IF (DABS(SA).LE.DABS(SB)) CS=SB/F0
-           DO 20 K=0,NM
-20            SJ(K)=CS*SJ(K)
-        ENDIF
-        DO 25 K=1,NM
-25         DJ(K)=SJ(K-1)-(K+1.0D0)*SJ(K)/X
-        RETURN
-        END
-
-        SUBROUTINE SPHY(N,X,NM,SY,DY)
+
+        integer,intent(in) :: n
+        real(dp),intent(in) :: x
+        integer,intent(out) :: nm
+        real(dp),dimension(0:n),intent(out) :: sj
+        real(dp),dimension(0:n),intent(out) :: dj
+
+        integer :: k,m
+        real(dp) :: sa,sb,f,f0,f1,cs
+
+        ! .3333333333333333d0 in original
+        real(dp),parameter :: one_third = 1.0_dp / 3.0_dp
+
+        nm=n
+        if (abs(x)<1.0e-100_dp) then
+           do k=0,n
+              sj(k)=0.0_dp
+              dj(k)=0.0_dp
+           end do
+           sj(0)=1.0_dp
+           if (n>0) then
+              dj(1)=one_third
+           endif
+           return
+        endif
+        sj(0)=sin(x)/x
+        dj(0)=(cos(x)-sin(x)/x)/x
+        if (n<1) then
+           return
+        endif
+        sj(1)=(sj(0)-cos(x))/x
+        if (n>=2) then
+           sa=sj(0)
+           sb=sj(1)
+           m=msta1(x,200)
+           if (m<n) then
+              nm=m
+           else
+              m=msta2(x,n,15)
+           endif
+           f=0.0_dp
+           f0=0.0_dp
+           f1=1.0_dp-100
```

Comment:
The original is `F1=1.0D0-100`, which is a bit confusing, but I think it's just `1-100` which is `-99`. Since `f1` is double precision, you can just write it as `1-100`.